transaction/interfaces.py

```diff
@@ -22,12 +22,19 @@ class ITransactionManager(Interface):
     """
 
     def begin():
-        """Begin a new transaction.
+        """Explicitly begin a new transaction.
 
         If an existing transaction is in progress, it will be aborted.
 
-        The newTransaction() method of registered synchronizers is called,
+        The ``newTransaction`` method of registered synchronizers is called,
         passing the new transaction object.
+
+        Note that transactions may be started implicitly without
+        calling ``begin``. In that case, ``newTransaction`` isn't
+        called because the transaction manager doesn't know when to
+        call it.  The transaction is likely to have begin long before
```

Comment:
get doesn't really create a transaction. It creates an object that represents a transaction.  As you point out, they are separate.

begin creates a transaction object as well.

begin is an application's opportunity to play nice and let everyone know what's going on.

I'm not proud of the implicit transaction model. If I had it do do over, I'd do it differently. (I think I did it this way originally based on experience with Ingres and/or Oracle.)

I'll also note that implicit behavior is useful for informal usage.  In fact, I could see use for a sort of an auto-commit mode that only allowed reads, although  I'm not sure if it could be done safely (because it could lead to invalidation of objects mid-computation).

In any case, allowing an explicit mode is an important first step. (And outside the scope of this pr. :))